File: app/src/test/java/com/mgaetan89/showsrage/model/ShowsFilters_State_GetViewIdForStateTest.java
package com.mgaetan89.showsrage.model;

import com.mgaetan89.showsrage.R;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class ShowsFilters_State_GetViewIdForStateTest {
	@Parameterized.Parameter(0)
	public ShowsFilters.State state;

	@Parameterized.Parameter(1)
	public int viewId;

	@Test
	public void getViewIdForState() {
		assertThat(ShowsFilters.State.Companion.getViewIdForState(this.state)).isEqualTo(this.viewId);
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{ShowsFilters.State.ACTIVE, R.id.filter_active},
				{ShowsFilters.State.ALL, R.id.filter_all},
				{ShowsFilters.State.PAUSED, R.id.filter_paused},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/model/ShowsFilters_State_GetViewIdForStateTest.kt
package com.mgaetan89.showsrage.model

import com.mgaetan89.showsrage.R
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class ShowsFilters_State_GetViewIdForStateTest(val state: ShowsFilters.State, val viewId: Int) {
    @Test
    fun getViewIdForState() {
        assertThat(ShowsFilters.State.getViewIdForState(this.state)).isEqualTo(this.viewId)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any>> {
            return listOf(
                    arrayOf(ShowsFilters.State.ACTIVE, R.id.filter_active),
                    arrayOf(ShowsFilters.State.ALL, R.id.filter_all),
                    arrayOf(ShowsFilters.State.PAUSED, R.id.filter_paused)
            )
        }
    }
}
